### PR TITLE
Splitting up CI matrix for pip llnl

### PR DESCRIPTION
**Pull request by MishaZakharchanka**
This PR focuses on the pip CI for LLNL machines. The functionality of the tests remains the same, but the jobs will now be listed individually on gitlab. Additionally, there was organizational work done in an attempt to make the file more readable and maintainable.

### CODE REVIEW

**Comment by MishaZakharchanka** [comment]
Do we want to add Lassen back to the machines that we are testing on?

**Comment by mtitov** [comment]
> Do we want to add Lassen back to the machines that we are testing on?

Yes.

And can you please investigate having a list of tags for `$RUNNER_TYPE`, since we need to add tag `slurm` for Slurm batch-runners (e.g., `RUNNER_TYPE: [ "batch", "slurm" ]`?)

File: .gitlab/llnl-ci-pip.yml [hunk 1]
```diff
@@ -102,30 +67,60 @@ pip-tests:
...
+pip_build_lassen:
+  variables:
+    RUNNER_TYPE: "shell"
+  needs: [pip_setup_lassen]
+  extends: [.on_lassen, .job_tags, .pip_build]
+
+pip_test_lassen:
+  variables:
+    RUNNER_TYPE: "batch"
+    LLNL_SLURM_SCHEDULER_PARAMETERS: "--nodes=1 -t 30"
```
Comment (mtitov) [inline .gitlab/llnl-ci-pip.yml:124]: Lassen doesn't have this variable `LLNL_SLURM_SCHEDULER_PARAMETERS`, but have the following ones
```
LLNL_LSF_SCHEDULER_PARAMETERS: "-nnodes 1 -W 30"
LSB_JOB_STARTER: "ENVIRONMENT=BATCH /usr/tcetmp/bin/bsub_job_starter %USRCMD"
ENVIRONMENT: "BATCH"
```

Comment (MishaZakharchanka) [reply .gitlab/llnl-ci-pip.yml:124]: Copy-pasting issues, thanks for catching

**Comment by MishaZakharchanka** [comment]
> And can you please investigate having a list of tags for `$RUNNER_TYPE`, since we need to add tag `slurm` for Slurm batch-runners (e.g., `RUNNER_TYPE: [ "batch", "slurm" ]`?)

Its looks like variables are only allowed to be strings, so I would write a comment to change 'batch' to 'slurm' either 6/1 or when the machines change schedulers to flux.

File: .gitlab/llnl-ci-pip.yml [hunk 2]
```diff
@@ -39,53 +41,16 @@ pip-env-setup:
...
-      #   RUNNER_TYPE: "batch"
-  tags:
-    - $HOST
-    - $RUNNER_TYPE
   script:
     - source ${PIP_WORK_DIR}/${VENV_ENV_NAME}/bin/activate
     - pip install --no-cache-dir -r ${PIP_REQUIREMENTS}
-    - *finalize
+    - chgrp -fR exaworks ${PIP_WORK_DIR} || true
+    - chmod -fR 02770 ${PIP_WORK_DIR}    || true
```
Comment (mtitov) [inline .gitlab/llnl-ci-pip.yml:50]: let's keep it within `finalize` and then should be called at the end for both `.pip_env_setup` and `.pip_build`

File: .gitlab/llnl-ci-pip.yml [hunk 3]
```diff
@@ -67,6 +67,11 @@ stages:
     - python3 ci/tests/test.py -e
     - deactivate
 
+.final_steps:
+  script: &finalize
+    - chgrp -fR exaworks ${PIP_WORK_DIR} || true
+    - chmod -fR 02770 ${PIP_WORK_DIR}    || true
+
```
Comment (mtitov) [inline .gitlab/llnl-ci-pip.yml:74]: please move it to the top

**Review by mtitov — APPROVED** [review]
LGTM!

**Pull request merged**